node/src/components/consensus/protocols/highway.rs

```diff
@@ -403,11 +404,32 @@ where
                         )];
                     }
                 };
+                let is_faulty = match pvv.inner().signed_wire_unit() {
+                    Some(signed_wire_unit) => self
+                        .highway
+                        .state()
+                        .is_faulty(signed_wire_unit.wire_unit.creator),
+                    None => false,
+                };
+
                 match pvv.timestamp() {
                     Some(timestamp) if timestamp > Timestamp::now() => {
-                        self.store_vertex_for_addition_later(timestamp, sender, pvv)
+                        // If it's not from an equivocator and from the future, add to queue
+                        if !is_faulty {
+                            self.store_vertex_for_addition_later(timestamp, sender, pvv)
+                        } else {
+                            return vec![];
```

Comment:
Is `return` required?

Reply:
Fixed in https://github.com/CasperLabs/casper-node/pull/521/commits/ea3e50e62089dcd7700d782bb5baf0767981cfa1